code/modules/vending/mail.dm

```diff
@@ -0,0 +1,300 @@
+#define STATE_SORTING "sorting"
+#define STATE_IDLE "idle"
+#define STATE_YES "yes"
+#define STATE_NO "no"
+#define MAIL_CAPACITY 100
+
+/obj/machinery/mailsorter
+	name = "mail sorter"
+	desc = "A large mail sorting unit. Sorting mail since 1987!"
+	icon = 'icons/obj/machines/mailsorter.dmi'
+	icon_state = "mailsorter"
+	base_icon_state = "mailsorter"
+	layer = BELOW_OBJ_LAYER
+	density = TRUE
+	max_integrity = 300
+	integrity_failure = 0.33
+	req_access = list(ACCESS_CARGO)
+	circuit = /obj/item/circuitboard/machine/mailsorter
+
+	var/light_mask = "mailsorter-light-mask"
+	var/panel_type = "panel"
+
+	/// What the machine is currently doing. Can be "sorting", "idle", "yes", "no".
+	var/currentstate = STATE_IDLE
+	/// List of all mail that's inside the mailbox.
+	var/list/mail_list = list()
+	/// The direction in which the mail will be unloaded.
+	var/output_dir = SOUTH
+	/// List of the departments to sort the mail for.
+	var/static/list/sorting_departments = list(
+		DEPARTMENT_ENGINEERING,
+		DEPARTMENT_SECURITY,
+		DEPARTMENT_MEDICAL,
+		DEPARTMENT_SCIENCE,
+		DEPARTMENT_CARGO,
+		DEPARTMENT_SERVICE,
+		DEPARTMENT_COMMAND,
+	)
+	var/static/list/choices = list(
+		"Eject" = icon('icons/hud/radial.dmi', "radial_eject"),
+		"Dump" = icon('icons/hud/radial.dmi', "mail_dump"),
+		"Sort" = icon('icons/hud/radial.dmi', "mail_sort"),
+	)
+
+/obj/machinery/mailsorter/proc/get_unload_turf()
+	return get_step(src, output_dir)
+
+/obj/machinery/mailsorter/screwdriver_act(mob/living/user, obj/item/tool)
+	default_deconstruction_screwdriver(user, "[base_icon_state]-off", base_icon_state, tool)
+	update_appearance()
```

Comment:
What is this call doing? ``default_deconstruction_screwdriver`` updates icons, is this for overlays? If so please add ``UPDATE_OVERLAYS`` arg

Reply:
maaaan...

it adds a panel. is panel considered an overlay?